controllers/awsmachine_controller.go

```diff
@@ -384,7 +427,46 @@ func (r *AWSMachineReconciler) getOrCreate(scope *scope.MachineScope, ec2svc ser
 	if instance == nil {
 		scope.Info("Creating EC2 instance")
 		// Create a new AWSMachine instance if we couldn't find a running instance.
-		instance, err = ec2svc.CreateInstance(scope)
+
+		// Set userdata.
+		var userData []byte
+
+		rawUserData, err := scope.GetRawBootstrapData()
+		if err != nil {
+			r.Recorder.Eventf(scope.AWSMachine, corev1.EventTypeWarning, "FailedGetBootstrapData", err.Error())
+			return nil, err
+		}
+
+		if scope.IsCloudInitEncrypted() {
+			compressedUserData, err := userdata.GzipBytes(rawUserData)
+			if err != nil {
+				return nil, err
+			}
+			// Do an initial check in case manager was terminated prematurely
+			secretARN := scope.GetSecretARN()
+			if secretARN == nil {
+				secretARN, err = secretSvc.Create(scope, compressedUserData)
+				if err != nil {
+					r.Recorder.Eventf(scope.AWSMachine, corev1.EventTypeWarning, "FailedCreatingSecret", err.Error())
+					return nil, err
+				}
+				scope.SetSecretARN(secretARN)
+			}
+			// Register the Secret ARN immediately to avoid orphaning AWS resources on delete
+			if err := scope.PatchObject(); err != nil {
+				return nil, err
+			}
+			encryptedCloudInit, err := secretsmanager.GenerateCloudInitMIMEDocument(*secretARN, scope.AWSCluster.Spec.Region)
+			if err != nil {
+				r.Recorder.Eventf(scope.AWSMachine, corev1.EventTypeWarning, "FailedGeneratingCloudInit", err.Error())
```

Comment:
(here & elsewhere) the event types are typically "Failed | Successful" + present tense verb + optional other words.